app/screens/sidebar/cityFilter/CityFilterContainer.js

```diff
@@ -0,0 +1,32 @@
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+import uiActions from 'actions/uiActions';
+import selector from './cityFilterSelector';
+
+export function UnconnectedCityFilterContainer(props) {
+  return (
+    <select className="city-filter" value={props.selected} onChange={props.onSelect}>
+      <option value="">Kaikki kaupungit</option>
+      {props.cities.map(city => (
+        <option key={city} value={city}>{city}</option>
+      ))}
+    </select>
+  );
+}
+
+UnconnectedCityFilterContainer.propTypes = {
+  cities: PropTypes.arrayOf(PropTypes.string).isRequired,
+  onSelect: PropTypes.func.isRequired,
+  selected: PropTypes.string.isRequired,
+};
+
+const actions = {
+  onSelect: (event) => {
+    const target = event.target;
+    const value = target.options[target.selectedIndex].value;
```

Comment:
- [ ] Is there any reason for not just using `const value = event.target.value;`